File: Algo_Method/calc_bits/ch10/Q2/Q2.rb
n, v = gets.chomp.split.map(&:to_i)
a = gets.chomp.split.map(&:to_i)
can_make = false
(1<<n).times do |bits|
    sum = 0
    n.times do |i| sum += a[i] if bits & (1<<i) != 0
    end
    can_make = true if sum == v
end
puts can_make ? "Yes" : "No"

File: Algo_Method/calc_bits/ch10/Q2/Q2.py
n, v = map(int, input().split())
a = list(map(int, input().split()))
can_make = False
for bits in range(1<<n):
    sum = 0
    for i in range(n):
        if bits & (1<<i): sum += a[i]
    if sum == v: can_make = True
print("Yes" if can_make else "No")